opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingFilter.java

```diff
@@ -59,7 +57,8 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
                 spanBuilder.asChildOf(extractedSpanContext);
             }
 
-            Span span = spanBuilder.start();
+            Span span = spanBuilder.startManual();
+            tracer.makeActive(span);
```

Comment:
This is ugly and the reason for this is that in async requests the thread in response filter is different and active span cannot be deactivated/finished: https://github.com/opentracing/opentracing-java/blob/master/opentracing-util/src/main/java/io/opentracing/util/ThreadLocalActiveSpan.java#L50.  

The most ideal solution would be to instrument `ExecutorService`/or something like `onAsync` but I could not find a way how to do it in jax-rs.

Other options:
1. do not activate at all (not wanted)
2. create "manual" span and activate explicitly. 
3. activate only for sync requests. (this is Brave doing https://github.com/pavolloffay/brave/blob/629a7da8d8e650124918b0e613b68bedd39cef34/instrumentation/jaxrs2/src/main/java/brave/jaxrs2/TracingContainerFilter.java#L74-L74)

cc @jpkrohling @objectiser